python-threatexchange/threatexchange/extensions/video_vpdq/vpdq_util.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import vpdq
+import json
+from json import JSONEncoder
+import typing as t
+import pathlib
+from dataclasses import dataclass
+
+QUALITY = "quality"
+HASH = "hash"
+TIMESTAMP = "timestamp"
+VPDQ_TIMESTAMP_PRECISION = 3
+
+
+@dataclass
+class VPDQMatchResult:
+    """Data class for VPDQ match result"""
+
+    query_match_percent: float = 0.0
+    compared_match_percent: float = 0.0
+
+
+def vpdq_to_json(vpdq_features: t.List[vpdq.VpdqFeature]) -> str:
+    """Convert from VPDQ features to json object and return the json object as a str"""
+    data: t.Dict[str, t.Dict] = {}
+    for feature in vpdq_features:
+        frame_number = feature.frame_number
+        data[frame_number] = {}
+        data[frame_number][QUALITY] = feature.quality
+        data[frame_number][HASH] = feature.hash
+        # VPDQ feature's timestamp is round to 3 decimals
+        data[frame_number][TIMESTAMP] = round(
+            feature.timestamp, VPDQ_TIMESTAMP_PRECISION
+        )
```

Comment:
As a followup, I think we do need to make this serialization less verbose. With another signal formats we "accepted" multiple formats but only produced one by default.